distsql: tpch query 7 results in `context cancelled` error

When running ([execution plan](https://raduberinde.github.io/decode.html?eJzcl19v2jAUxd_3KSK_AKsnYZPwJ9Iks7bbmAZstH3ZhKaUeBApJMgxUquJ7z6RtGr-dPc2pHRR30KSX3yPfc-x-UOC0JUTZy0jYv8kjFDCCSUdMqdko8KFjKJQ7R8lL47cG2K3KfGCzVYnt7WnfUlssg1C5UolXUKJK7Xj-fvn892ckkWoJLEfXv0SesFMOq5UmXfJRnlrR90K3wukp-WaUDLdatsQjIoOFV0qelQwRuY7SsKtLpRwfWusnGiV_ajYvz_f0X8IeKS8S-fal1B9gaO9MCCUfPR8LZVtCG6MJkazcTqcDM-GDdu2Ly5no8knajTG0x_D8YfR96vzh9uttDCek_MqSsXbgh7UOZ-daLXvnrzc6cTw5W_dbL1X3nKlm9my4-4xS7ZNB2yb_yVQ8HuJotPK-cOkovus3qiBSJYRaVJhvRqR7DGR_H4lrTjs-lQMyqqlwqqpYJpeWCqsveyZDNw4l_qGGBjyRitnoZuNW-modDiJbssQpvHWaDLbts_OT0fj4VfjnSGs1hPQKhE7XC6VXDo6zMmKO5GKHqFkdHY-ubxbxrtrnrqOjXpxNW4KM3XXSl13U9e9p1d70DI9s54KM3sRKp1vFsFOqOAnVHROwE-zA_dF_1fkPnK-6B1lsPtNOF6bCNpJMVPzgqNZHRx97JyuhciXzOl6CK5nTiMurFlO84qZEW03G9-Lnz7YqnyLFYPj0MIOSs6qs3CM5Kxa02Ib6XBdbWXgqmqQ58f5c1EzkcfZtGoh8iU3rXoIruemhYRNzTYt5MA-k9EmDCKZX9HM9-aUSHcpk2aIwq1ayG8qXMQfT35O47fjDd2VkU6edpMfoyB5tC_r6TAzq9C8A9MsT7fTNALzEjDPwx24blaFNmHYhCccKRymOa9CWzBslWizkjBD5gymOTJnXZhGxobpXhWYITPeA1u8n4HbebgPwgPYXAMQZm14aFaIhQzOuwgO50IbCZVCMEAGK4vnHFYWtxAaThbEYwiNmQzBMZcxJJqw0WEc8RlCY0ZjhYDJ9jvWsAWbl_EaK9gcmriyeK8SzbCGhVMGs2ohZspZFcZRq8I4opwjRx9WCUfWnBcSstQhAsGxUwSCI17jcEIiGYfQWMYhOJZxHIlIbHQYRzIOobGM44WIzG7KFpxxHI7IfMbNd2_-BgAA___HJ6WX))

```
SELECT
    supp_nation,
    cust_nation,
    l_year,
    SUM(volume) AS revenue
FROM
    (
        SELECT
            n1.n_name AS supp_nation,
            n2.n_name AS cust_nation,
            EXTRACT(year FROM l_shipdate) AS l_year,
            l_extendedprice * (1 - l_discount) AS volume
        FROM
            supplier,
            lineitem,
            orders,
            customer,
            nation n1,
            nation n2
        WHERE
            s_suppkey = l_suppkey
            AND o_orderkey = l_orderkey
            AND c_custkey = o_custkey
            AND s_nationkey = n1.n_nationkey
            AND c_nationkey = n2.n_nationkey
            AND (
                (n1.n_name = 'MOZAMBIQUE' AND n2.n_name = 'CANADA')
                or (n1.n_name = 'CANADA' AND n2.n_name = 'MOZAMBIQUE')
            )
            AND l_shipdate BETWEEN DATE '1995-01-01' AND DATE '1996-12-31'
    ) AS shipping
GROUP BY
    supp_nation,
    cust_nation,
    l_year
ORDER BY
    supp_nation,
    cust_nation,
    l_year;
```

against a 3-node local cluster sometimes crashes all three servers with
```
panic: too many Close() calls

goroutine 124280 [running]:
panic(0x5371480, 0xc428fe20b0)
    /usr/local/Cellar/go/1.7.4_2/libexec/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*MultiplexedRowChannel).Close(0xc43d2b8fc0, 0x5f41e60, 0xc458deca20)
    /Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:195 +0x21a
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*flowRegistry).finishInboundStreamLocked(0xc42023e020, 0xc4430d1520)
    /Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow_registry.go:274 +0xbf
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*flowRegistry).RegisterFlow.func1()
    /Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow_registry.go:157 +0x2f6
created by time.goFunc
    /usr/local/Cellar/go/1.7.4_2/libexec/src/time/sleep.go:154 +0x44
```

And sometimes only crashes the gateway server with
```
panic: send on closed channel

goroutine 1280 [running]:
panic(0x53e93e0, 0xc439ee7400)
    /usr/local/Cellar/go/1.7.4_2/libexec/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*RowChannel).PushRow(0xc420225240, 0xc4202c43b8, 0x7, 0x7, 0x0)
    /Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:121 +0xc7
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*MultiplexedRowChannel).PushRow(0xc420225240, 0xc4202c43b8, 0x7, 0x7, 0xc449580000)
    /Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:181 +0x4b
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*procOutputHelper).emitRow(0xc4202c4388, 0x8316fb8, 0xc4211a4480, 0xc449580000, 0x7, 0x7, 0x5f60480)
    /Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:197 +0x26c
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*aggregator).Run(0xc4202c3200, 0x8316fb8, 0xc4211a4480, 0xc4201dede8)
    /Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/aggregator.go:162 +0x432
created by github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).Start
    /Users/asubiotto/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:304 +0x329```

--- a/pkg/sql/distsqlrun/flow_registry.go
+++ b/pkg/sql/distsqlrun/flow_registry.go
@@ -270,10 +270,6 @@ func (fr *flowRegistry) finishInboundStreamLocked(is *inboundStreamInfo) {
 		panic("double finish")
 	}
 
-	if is.timedOut {
-		is.receiver.Close(errors.Errorf("inbound stream timed out"))
-	}
-
 	is.finished = true
 	is.waitGroup.Done()
 }

